Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+    m_dir = AxisDirection::AxisY;
+  } else if ((localDirVector - Vector3::UnitZ()).norm() < tolerance) {
+    m_dir = AxisDirection::AxisZ;
+  } else {
+    throw std::invalid_argument("CuboidVolumeStack: Invalid axis direction");
+  }
+
+  auto dirIdx = static_cast<std::size_t>(m_dir);
+  m_dirOrth1 = static_cast<AxisDirection>((dirIdx + 1) % 3);
+  m_dirOrth2 = static_cast<AxisDirection>((dirIdx + 2) % 3);
+
+  initializeOuterVolume(strategy, logger);
+}
+
+Volume& CuboidVolumeStack::initialVolume(const std::vector<Volume*>& volumes) {
+  if (volumes.empty()) {
+    throw std::invalid_argument(
+        "CuboidVolumeStack requires at least one volume");
+  }
+  return *volumes.front();
+}
+
+void CuboidVolumeStack::initializeOuterVolume(AttachmentStrategy strategy,
+                                              const Logger& logger) {
+  ACTS_DEBUG("Creating CuboidVolumeStack from " << m_volumes.size()
+                                                << " volumes in direction "
+                                                << axisDirectionName(m_dir));
+  if (m_volumes.empty()) {
+    throw std::invalid_argument(
+        "CuboidVolumeStack requires at least one volume");
+  }
+
+  if (m_dir != Acts::AxisDirection::AxisX &&
+      m_dir != Acts::AxisDirection::AxisY &&
+      m_dir != Acts::AxisDirection::AxisZ) {
+    throw std::invalid_argument(axisDirectionName(m_dir) +
+                                " is not supported ");
+  }
+
+  // For alignment check, we have to pick one of the volumes as the base
+  m_groupTransform = m_volumes.front()->transform();
+  ACTS_VERBOSE("Initial group transform is:\n" << m_groupTransform.matrix());
+
+  std::vector<VolumeTuple> volumeTuples;
+  volumeTuples.reserve(m_volumes.size());
+
+  for (const auto& volume : m_volumes) {
+    const auto* cuboidBounds =
+        dynamic_cast<const CuboidVolumeBounds*>(&volume->volumeBounds());
+    if (cuboidBounds == nullptr) {
+      throw std::invalid_argument{
+          "CuboidVolumeStack requires all volumes to "
+          "have CuboidVolumeBounds"};
+    }
+
+    volumeTuples.emplace_back(*volume, m_groupTransform);
+  }
+
+  ACTS_DEBUG("*** Initial volume configuration:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::DEBUG);
+
+  if (m_volumes.size() == 1) {
+    ACTS_VERBOSE("Only one volume, returning");
+    setTransform(m_volumes.front()->transform());
+    const auto* bounds = dynamic_cast<const CuboidVolumeBounds*>(
+        &m_volumes.front()->volumeBounds());
+    assert(bounds != nullptr && "Volume bounds are not cuboid bounds");
+    Volume::update(std::make_shared<CuboidVolumeBounds>(*bounds), std::nullopt,
+                   logger);
+    ACTS_VERBOSE("Transform is now: " << m_transform.matrix());
+    return;
+  }
+
+  ACTS_VERBOSE("Checking volume alignment");
+  checkVolumeAlignment(volumeTuples, logger);
+
+  auto dirIdx = static_cast<std::size_t>(m_dir);
```

Comment:
This currently works out (which might be what @asalzburger originally intended), but I'm a bit worried to always assume this. I'd probably just do this with a `switch` to be sure.